src/lib/RelayDecayLib.sol

```diff
@@ -48,38 +49,53 @@ library RelayDecayLib {
 
     /// @notice returns a decayed input array using the given decay spec and times
     /// @param inputs The input array to decay
-    /// @param decayStartTime The time to start decaying
-    /// @param decayEndTime The time to end decaying
     /// @return result a decayed input array
-    function decay(InputTokenWithRecipient[] memory inputs, uint256 decayStartTime, uint256 decayEndTime)
-        internal
-        view
-        returns (InputTokenWithRecipient[] memory result)
-    {
+    function decay(RelayInput[] memory inputs) internal view returns (InputTokenWithRecipient[] memory result) {
```

Comment:
interesting so each input can now have different decay start and end? seems like thats a pretty significant change from the older design?

same for outputs?

Reply:
slightly less of a change would just be that the inputs all share a decay and the outputs all share a decay.. benefit being maybe slightlyyy more general/easier to parameterize but negative being way less flexible?

Reply:
Hmm yeah so ideally the input for the swap will have equal startAmount and endAmount, so no decay (since the swap is discrete), and the input offered for gas will have decay + its own parametrization times.

Reply:
Thinking a bit more, and I think having all inputs share a decay + outputs share a decay is best. Way easier integration for backend services and we don't actually need that finegrained of control